### PR TITLE
Add auto-exit to cded export feed

### PR DESCRIPTION
**Pull request by gigxz**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Add `auto_exit` key to the export feed, as requested.

https://github.com/open-path/Green-River/issues/6172

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/models/hmis/hud/enrollment.rb [hunk 1]
```diff
@@ -193,7 +193,8 @@ def set_hud_project_id_from_project_pk_unless_wip
     joins(:client).merge(Hmis::Hud::Client.age_group(start_age: start_age, end_age: end_age))
   end
 
-  scope :exited, -> { left_outer_joins(:exit).where(ex_t[:ExitDate].not_eq(nil)) }
+  scope :exited, -> { joins(:exit).where(ex_t[:ExitDate].not_eq(nil)) }
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/enrollment.rb:196]: Unrelated fix - I see no need for left join here

File: drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/cde_export.rb [hunk 2]
```diff
@@ -27,6 +30,53 @@ def run!
...
+      # Include Auto Exit flag for auto-exited Enrollments
+      Hmis::Hud::Enrollment.hmis.auto_exited.preload(:exit).each do |enrollment|
+        values = [
+          enrollment.exit.id,  # ResponseID
+          AUTO_EXIT_KEY,       # CustomFieldKey
+          'Enrollment',        # RecordType
+          enrollment.id,       # RecordId
+          'true',              # Response
+          enrollment.exit.auto_exited,  # DateCreated (passing the date that the auto-exit occurred)
+          enrollment.exit.DateUpdated,  # DateUpdated
```
Comment (gigxz) [inline drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/cde_export.rb:73]: A spec was not provided, but this should be sufficient. I did not include the actual `ExitDate` here because we don't easily have the Exit Date that was specified by the AutoExitJob. (Because the ExitDate could have changed after the auto-exit occurred).

File: drivers/hmis/app/models/hmis/hud/enrollment.rb [hunk 3]
```diff
@@ -193,7 +193,8 @@ def set_hud_project_id_from_project_pk_unless_wip
...
+  scope :auto_exited, -> { joins(:exit).where(ex_t[:ExitDate].not_eq(nil).and(ex_t[:auto_exited].not_eq(nil))) }
```
Comment (eanders) [inline drivers/hmis/app/models/hmis/hud/enrollment.rb]: This is fine as is, but could be written with a merge to allow for future changes to the exit `auto_exited` scope.
```suggestion
  scope :auto_exited, -> { joins(:exit).merge(Hmis::Hud::Exit.auto_exited.where.not(exit_date: nil) }
```

**Review by eanders — APPROVED** [review]

**Review by ttoomey — APPROVED** [review]
looks good!
